### PR TITLE
Feature/python api

### PR DESCRIPTION
**Pull request by yml**
This pull request fix a small bug in the add_plugin api when you pass a subclass of CMSPluginBase. The order matter.

I have also notice that I had to remove the save=False from this signature:
- plugin_base.insert_at(None, position=position, save=False)

However this might be due to the fact that mptt signature has change for this method. I am using a version of the cms that bundle a patch mptt.

The publish_page  method is probably not the best name for a method that toggle the state page.published attr.

Regards,
--yml

### CODE REVIEW

File: cms/api.py [hunk 1]
```diff
@@ -207,8 +207,8 @@ def add_plugin(placeholder, plugin_type, language, position='last-child', **data
     
     # validate and normalize plugin type
     if hasattr(plugin_type, '__module__') and  issubclass(plugin_type, CMSPluginBase):
-        plugin_type = plugin_type.__name__
         plugin_model = plugin_type.model
+        plugin_type = plugin_type.__name__
```
Comment (ojii) [inline cms/api.py:211]: Why this change?

File: cms/api.py [hunk 2]
```diff
@@ -219,12 +219,12 @@ def add_plugin(placeholder, plugin_type, language, position='last-child', **data
         
         
     plugin_base = CMSPlugin(
-        plugin_type='TextPlugin',
+        plugin_type=plugin_type,
```
Comment (ojii) [inline cms/api.py:222]: Oh wow...

File: cms/api.py [hunk 3]
```diff
@@ -219,12 +219,12 @@ def add_plugin(placeholder, plugin_type, language, position='last-child', **data
...
         placeholder=placeholder, 
         position=1, 
         language=language
     )
-    plugin_base.insert_at(None, position=position, save=False)
+    plugin_base.insert_at(None, position=position, save=False)  # save is causing trouble but it might be because I have tested with an older version of django-cms
```
Comment (ojii) [inline cms/api.py:227]: Yes the save is because of the new MPTT, please remove this comment

File: cms/api.py (hunk 1 above)
Comment (yml) [inline cms/api.py:211]: If you do it in the other order : plugin_model = plugin_type.model raise an exception because the line before you put a string in the plugin_type.

Comment (ojii) [inline cms/api.py:211]: Wow, fail on my end...
